### PR TITLE
Always create ECR Pull Through Cache Policy

### PR DESCRIPTION
**Pull request by santhoshratala**
- Removed the input variable "configure_ecr_pull_through_policy"
- Added resource to attach existing policy to iam worker role

### CODE REVIEW

**Comment by lawliet89** [comment]
Hey @santhoshratala can you fix the conflicts first, please.

**Review by niroz89 — COMMENTED** [review]
@santhoshratala removed var `configure_ecr_pull_through_policy` still in use.

File: modules/essentials/variables.tf [hunk 1]
```diff
@@ -313,10 +313,10 @@ variable "configure_ecr_pull_through" {
   default     = true
 }
 
-variable "configure_ecr_pull_through_policy" {
+variable "ecr_pull_through_existing_policy_arn" {
   description = "Configure ECR Pull Through Cache IAM policy."
-  type        = bool
-  default     = true
+  type        = list(string)
```
Comment (lawliet89) [inline modules/essentials/variables.tf:318]: Just make this a set from the beginning. Then you don't have to use `toset`.

**Comment by santhoshratala** [comment]
rolling back changes made in https://github.com/SPHTech-Platform/terraform-aws-eks/pull/20

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**